core/src/main/java/org/apache/iceberg/rest/auth/OAuth2Manager.java

```diff
@@ -273,15 +273,15 @@ protected OAuth2Util.AuthSession newSessionFromTokenResponse(
   }
 
   private static void warnIfDeprecatedTokenEndpointUsed(Map<String, String> properties) {
-    if (usesDeprecatedTokenEndpoint(properties)) {
+    if (!properties.containsKey(OAuth2Properties.OAUTH2_SERVER_URI)) {
       String credential = properties.get(OAuth2Properties.CREDENTIAL);
       String initToken = properties.get(OAuth2Properties.TOKEN);
       boolean hasCredential = credential != null && !credential.isEmpty();
       boolean hasInitToken = initToken != null;
       if (hasInitToken || hasCredential) {
         LOG.warn(
             "Iceberg REST client is missing the OAuth2 server URI configuration and defaults to {}/{}. "
-                + "This automatic fallback will be removed in a future Iceberg release."
+                + "This automatic fallback will be removed in a future Iceberg release. "
```

Comment:
nit: is this extra space required? 
```suggestion
                + "This automatic fallback will be removed in a future Iceberg release."
```

Reply:
I think so? As it's concatenated to the next sentence below.